packages/react/src/table/TableScroller.tsx

```diff
@@ -204,140 +215,212 @@ export function TableScroller({ children }: TableScrollerProps) {
 		}
 	};
 
+	const scrollerHeight = scrollerRef.current?.scrollHeight;
+	const isScrolledToEnd =
+		scrollerRef.current &&
+		scrollerRef.current.scrollLeft ===
+			scrollerRef.current.scrollWidth - scrollerRef.current.clientWidth;
+
 	return (
-		<Stack
-			gap={0.5}
-			css={{
-				position: 'relative',
-				width: '100%',
+		<ScrollerContext.Provider
+			value={{
+				setOverlayOffsets,
 			}}
 		>
-			<Box
+			<Stack
+				maxWidth="xs"
+				gap={0.5}
 				css={{
-					msOverflowStyle: 'none',
-					overflowX: 'auto',
-					scrollbarWidth: 'none',
-					WebkitOverflowScrolling: 'touch',
+					containerType: 'inline-size',
+					position: 'relative',
 					width: '100%',
-					'&::-webkit-scrollbar, &::-webkit-scrollbar-thumb, &::-webkit-scrollbar-track':
-						{
-							display: 'none',
-						},
 				}}
-				focusRingFor="keyboard"
-				onScroll={repositionThumb}
-				ref={scrollerRef}
-				tabIndex={0}
-			>
-				{children}
-			</Box>
-			<Flex
-				background="body"
-				alignItems="center"
-				css={{
-					bottom: 0,
-					display: thumbWidthRatio === 1 ? 'none' : undefined,
-					left: 0,
-					position: 'sticky',
-					right: 0,
-				}}
-				flexWrap="nowrap"
-				gap={0.25}
 			>
 				<Box
-					aria-hidden
-					as="button"
+					data-table-left-overlay="true"
+					css={{
+						left: overlayOffsets?.left,
+						pointerEvents: 'none',
+						position: 'fixed',
+						zIndex: SCROLL_OVERLAY_Z_INDEX,
+						...(scrollerHeight && { height: `${scrollerHeight / 16}rem` }),
+
+						// Switch between single line at full scroll and overlay shadow at partial scroll
+						...(scrollerRef.current?.scrollLeft === 0
+							? {
+									background: boxPalette.borderMuted,
+									width: '1px',
```

Comment:
I don't think we should be applying this border if there are no frozen columns 
<img width="64" alt="image" src="https://github.com/user-attachments/assets/36c3e3fe-de7d-4a96-9168-c06341f49493">

Reply:
Why is that? Doesn't it signify the hard edge of scroll, as opposed to the frozen column?

Reply:
I don't think so. I think it only signifies the edge of a fixed column that isn't scrolled. The table should keep its "blank" edges in its normal 0 or fully scrolled states